aws_topology/stackstate_checks/aws_topology/resources/stepfunction.py

```diff
@@ -0,0 +1,344 @@
+
+    def get_function_reference(self, reference):
+        fn_arn = ''
+        if isinstance(reference, string_types):
+            parts = reference.split(':', -1)
+            if isinstance(parts, list):
+                if len(parts) == 7:
+                    # plain lambda arn
+                    fn_arn = ':'.join(parts)
+                elif len(parts) == 8 and (parts[7].isdigit() or parts[7].lower() == '$latest'):
+                    # TODO lambda versions are not yet supported in StackState so omit version
+                    fn_arn = ':'.join(parts[0:7])
+                elif len(parts) == 8:
+                    # lambda alias arn
+                    fn_arn = ':'.join(parts)
+                elif len(parts) >= 2 and len(parts) <= 4:
+                    # partial lambda arn OR nameonly + version/alias
+                    end = len(parts)
+                    if parts[-1].isdigit() or parts[-1].lower() == '$latest':
+                        end = end - 1
+                    start = 0
+                    if end - start > 2:
+                        start = 2
+                    fn_arn = self.agent.create_arn(
+                        'AWS::Lambda::Function',
+                        self.location_info,
+                        ':'.join(parts[start:end])
+                    )
+                if len(parts) == 1:
+                    # only lambda name
+                    fn_arn = self.agent.create_arn(
+                        'AWS::Lambda::Function',
+                        self.location_info,
+                        ':'.join(parts)
+                    )
+        return fn_arn
+
+    def process_task_state(self, state_arn, state):
+        # TODO sense intrinsics used in any integration parameter -> don't make relation
+        # TODO resource end part is invokation type put it in the state
+        # TODO support AWS Batch / AWS Glue (DataBrew) / EMR / EMR on EKS / EKS / CodeBuild
+        # TODO decide on (Athena / SageMaker)
+        partition = get_partition_name(self.location_info['Location']['AwsRegion'])
+        resource = state.get('Resource') or ''
+        parameters = state.get('Parameters') or {}
+        parts = resource.split(":", -1)
+        if len(parts) > 6:
+            integration_type = ':'.join(parts[6:])
+            if integration_type:
+                state["IntegrationSubType"] = integration_type
+        if resource.startswith('arn:{}:states:::lambda:'.format(partition)):
+            state["IntegrationType"] = "lambda"
+            fn_arn = self.get_function_reference(parameters.get('FunctionName'))
+            if fn_arn:
+                self.agent.relation(state_arn, fn_arn, 'uses service', {})
+            else:
+                self.agent.warning('Could not make lambda relation of {}'.format(
+                    parameters.get('FunctionName')
+                ))
+        elif resource.startswith('arn:{}:lambda:'.format(partition)):
+            state["IntegrationType"] = "lambda"
```

Comment:
Looks like some duplicate code here with only the `get_function_reference` parameter changing, probably a good case to separate into it's own function

Reply:
changed a bit :-)